src/onnxruntime.cc

```diff
@@ -1482,9 +1512,30 @@ ModelInstanceState::ValidateOutputs()
 
     // The batch output shape doesn't necessarily match the model
     if (model_state_->FindBatchOutput(io_name) == nullptr) {
-      RETURN_IF_ERROR(CompareDimsSupported(
-          model_state_->Name(), io_name, iit->second.dims_, dims,
-          model_state_->MaxBatchSize(), true /* compare_exact */));
+      // if max_batch_size == 0 and is a scalar tensor all the
+      // dimensions specified must be equal to 1
+      if (model_state_->MaxBatchSize() > 0 || iit->second.dims_.size() > 0) {
```

Comment:
Summarizing offline discussion, I'd rather see `MaxBatchSize()` removed from this condition of "is_scalar" if not necesary. Maybe something like this:
```cpp
const bool is_scalar = iit->second.dims_.size() == 0;
if (is_scalar) {
  if std::any_of(dims.begin(), dims.end(), [](int dim) { dim != 1}) {
    // ERROR
  }
  scalar_outputs_[io_name] = dims;
} else {
  RETURN_IF_ERROR(CompareDimsSupported(...));
}
```

Reply:
fixed.